src/main/java/com/endava/internship/collections/StudentList.java

```
@@ -1,122 +1,241 @@
 package com.endava.internship.collections;
 
-import java.util.Collection;
-import java.util.Iterator;
-import java.util.List;
-import java.util.ListIterator;
+import lombok.ToString;
 
-public class StudentList implements List<Student> {
+import java.util.*;
+import java.util.function.Consumer;
+
+@ToString
+public class StudentList extends AbstractList<Student> implements List<Student> {
+
+    private int sizeOfArray = 0;
+    private static int CAPACITY = 10;
+    private static Student[] elements = new Student[CAPACITY];
+
+    public StudentList() {
+    }
+
+    public StudentList(int initialCapacity) {
+        CAPACITY = initialCapacity;
+        updateCapacity(elements, initialCapacity);
+    }
+
+    public StudentList(Collection<? extends Student> studentsCollection) {
+        addAll(studentsCollection);
+    }
 
     @Override
     public int size() {
-        //TODO
-        return 0;
+        return sizeOfArray;
     }
 
     @Override
     public boolean isEmpty() {
-        //TODO
-        return false;
+        return sizeOfArray == 0;
     }
 
     @Override
     public boolean contains(Object o) {
-        //TODO
-        return false;
+        return indexOf(o) != -1 ? true : false;
     }
 
     @Override
     public Iterator<Student> iterator() {
-        //TODO
-        return null;
+        return new Iterator<Student>() {
```

There are two types of iterator fail-safe and fail-fast (google it).
From documentation arraylist iterator should be fail-fast - https://docs.oracle.com/javase/8/docs/api/java/util/ArrayList.html#iterator--

That means it should throw an exception in case if collection was modified.